src/libp2p/Libp2p.Protocols.Pubsub/PubsubRouter.Topics.cs

```diff
@@ -82,34 +83,41 @@ public void Unsubscribe(string topicId)
 
     public void UnsubscribeAll()
     {
-        foreach (PeerId? peerId in fPeers.SelectMany(kv => kv.Value))
+        try
         {
-            Rpc msg = new Rpc().WithTopics([], topicState.Keys);
+            foreach (PeerId? peerId in fPeers.SelectMany(kv => kv.Value))
+            {
+                Rpc msg = new Rpc().WithTopics([], topicState.Keys);
 
-            peerState.GetValueOrDefault(peerId)?.Send(msg);
-        }
+                peerState.GetValueOrDefault(peerId)?.Send(msg);
+            }
 
-        Dictionary<PeerId, Rpc> peerMessages = new();
+            Dictionary<PeerId, Rpc> peerMessages = [];
 
-        foreach (PeerId? peerId in gPeers.SelectMany(kv => kv.Value))
-        {
-            (peerMessages[peerId] ??= new Rpc())
-                .WithTopics([], topicState.Keys);
-        }
-
-        foreach (KeyValuePair<string, HashSet<PeerId>> topicMesh in mesh)
-        {
-            foreach (PeerId peerId in topicMesh.Value)
+            foreach (PeerId? peerId in gPeers.SelectMany(kv => kv.Value))
             {
                 (peerMessages[peerId] ??= new Rpc())
-                   .Ensure(r => r.Control.Prune)
-                   .Add(new ControlPrune { TopicID = topicMesh.Key });
+                    .WithTopics([], topicState.Keys);
             }
-        }
 
-        foreach (KeyValuePair<PeerId, Rpc> peerMessage in peerMessages)
+            foreach (KeyValuePair<string, HashSet<PeerId>> topicMesh in mesh.ToDictionary())
```

Comment:
do we need to allocate a dictionary?
```suggestion
            foreach (KeyValuePair<string, HashSet<PeerId>> topicMesh in mesh)
```